lib/MusicBrainz/Server/Form/User/Register.pm

```diff
@@ -10,20 +10,45 @@ with 'MusicBrainz::Server::Form::Role::CSRFToken';
 
 has '+name' => ( default => 'register' );
 
+my $text_too_long_message = N_l('The value of this field cannot be longer than {max} characters, but you entered {actual}.');
+
 has_field 'username' => (
     type      => 'Text',
     required  => 1,
     maxlength => 64,
     validate_method => \&validate_username,
+    messages  => {
+        text_maxlength => $text_too_long_message,
+    },
+    localize_meth => sub {
+        my ($self, $message, @args) = @_;
+        
+        if ($message eq $text_too_long_message) {
+            return l($message, { max => $args[0], actual => $args[1] });
+        }
+
+        return l($message);
+    }
```

Comment:
I think it's worth moving this to a named subroutine so you can just do `localize_meth => \&localize_meth,` here and below.